### PR TITLE
Fix #257 uppercase attribute names will stay uppercased

### LINKED ISSUE #257: Error: Call to a member function isCollection() on null
I'm not sure that this is directly related to DunglasApiBundle but thought it'd be nice to keep a trace here.

Error origin: [ItemNormalizer#denormalize](https://github.com/dunglas/DunglasApiBundle/blob/master/JsonLd/Serializer/ItemNormalizer.php#L244). Note that, compared to the `normalize` function, there is no test for `!$type`.
Anyway, type should not be empty here, so I tracked back this error to [AttributeMetadataFactory#getAttributeMetadataFor](https://github.com/dunglas/DunglasApiBundle/blob/master/Mapping/Factory/AttributeMetadataFactory.php#L57), where the `reflectionProperty` was empty.
 
Weird because this property is declared the exact same way as the other, and as a string:

```php
    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=255)
     */
    private $name;

    /**
     * @var string
     *
     * @ORM\Column(name="SIRET", type="string", length=255, nullable=true)
     */
    private $SIRET;
``` 

The bug came from `SIRET` declared in uppercase:

`AttributeName` is `sIRET` but the `reflectionClass` gives `SIRET`:

```php
ReflectionClass {#713
  constants: []
  properties: array:7 [
    "SIRET" => ReflectionProperty {#885
      +name: "SIRET"
      +class: "\\\InternalOrganization"
      modifiers: "private"
      extra: array:1 [
        "docComment" => """
          /**\n
               * @var string\n
               *\n
               * @ORM\Column(name="SIRET", type="string", length=255, nullable=true)\n
               */
          """
      ]
    }
  ]
}
```

I know properties should not be uppercased and I've changed it, but this could be a bug. I let y'all be the judge :).

### PR DESCRIPTION
**Pull request by soyuka**
I've added some unit test, please let me know if I did something "the wrong way".

### CODE REVIEW

File: Tests/Mapping/Loader/ReflectionLoaderTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+class ReflectionLoaderTest extends \PHPUnit_Framework_TestCase
+{
+  
+    public function testUppercaseMethods() 
+    {
+        $testClass = new MyTestClass();
+
+        $reflectionClassProphecy = $this->prophesize('ReflectionClass');
```
Comment (sroze) [inline Tests/Mapping/Loader/ReflectionLoaderTest.php]: Why not instantiating directly the `ReflectionClass` ?

Comment (soyuka) [inline Tests/Mapping/Loader/ReflectionLoaderTest.php]: Duno, I used [this](https://github.com/dunglas/DunglasApiBundle/blob/master/Tests/Mapping/Loader/AnnotationLoaderTest.php) as an example and I probably got lost trying to understand what should call whom. It's a bit hard to write this kind of tests when you're not familiar with the ApiBundle internals, in my opinion.

**Comment by dunglas** [comment]
It looks good to me but 2.0 will use the new Symfony Property Info Component.
Can you:
- apply this fix to 1.0
- open a PR against symfony/symfony#15858 for 2.0?

**Comment by sroze** [comment]
@dunglas the PropertyInfo component do not `lcfirst` properties' names so no need to fix on it :)

**Comment by dunglas** [comment]
@soyuka can you reopen this PR against 1.0, rebase and squash your commits?

**Comment by soyuka** [comment]
This seems to be fixed in 1.0 ;)

**Pull request closed without merging**